osu.Game/Beatmaps/WorkingBeatmap.cs

```diff
@@ -79,100 +78,98 @@ protected virtual Track GetVirtualTrack(double emptyLength = 0)
         /// <returns>The applicable <see cref="IBeatmapConverter"/>.</returns>
         protected virtual IBeatmapConverter CreateBeatmapConverter(IBeatmap beatmap, Ruleset ruleset) => ruleset.CreateBeatmapConverter(beatmap);
 
-        public virtual IBeatmap GetPlayableBeatmap(IRulesetInfo ruleset, IReadOnlyList<Mod> mods = null, TimeSpan? timeout = null)
+        public virtual IBeatmap GetPlayableBeatmap(IRulesetInfo ruleset, IReadOnlyList<Mod> mods = null, CancellationToken? cancellationToken = null)
         {
-            using (var cancellationSource = createCancellationTokenSource(timeout))
-            {
-                mods ??= Array.Empty<Mod>();
+            var token = cancellationToken ?? new CancellationTokenSource(10000).Token;
```

Comment:
I've debated for a while as to whether to dispose of this `CancellationTokenSource`, but I left it be for now as I *think* it should be fine to leave cleaning it up to GC as it's not linked, so it shouldn't leak memory...? And cleaning it up may complicate code a touch. Not sure.

Reply:
Another case where having an overload with no cancellation token would work really well. Wondering if making `mods` a required parameter is a better path?

Reply:
Yeah, overloads would help here. But as you say, mods would probably have to become mandatory for practicality reasons alone (to reduce number of parameter combinations to add overloads for)

Reply:
So `new DifficultyCalculator(..., mods)`? I think that's fine.

Reply:
Mods is already there but has a default of `null`, allowing calls to `GetPlayableBeatmap(ruleset)` with no mods provided. The issue is that if we want two versions of the function - one with and one without the cancellationToken - the ordering becomes an issue (we'd want `cancellationToken` to be last but it can't be if there's an optional `mods` argument).

Options would be to make four overloads to work around this, or to make `mods` non-optional.

Reply:
It's fine as it is, imo.